Release/src/json/json_parsing.cpp

```diff
@@ -682,26 +722,31 @@ inline bool JSON_Parser<CharType>::handle_unescape_char(Token& token)
         case 't': token.string_val.push_back('\t'); return true;
         case 'u':
         {
-            // A four-hexdigit Unicode character.
-            // Transform into a 16 bit code point.
-            int decoded = 0;
-            for (int i = 0; i < 4; ++i)
+            int decoded = convert_unicode_to_code_point();
+            if (decoded == -1)
             {
-                ch = NextCharacter();
-                int ch_int = static_cast<int>(ch);
-                if (ch_int < 0 || ch_int > 127) return false;
-#ifdef _WIN32
-                const int isxdigitResult = _isxdigit_l(ch_int, utility::details::scoped_c_thread_locale::c_locale());
-#else
-                const int isxdigitResult = isxdigit(ch_int);
-#endif
-                if (!isxdigitResult) return false;
+                return false;
+            }
+
+            // handle multi-block characters that start with a high-surrogate
+            if (decoded > H_SURROGATE_START && decoded < H_SURROGATE_END)
```

Comment:
While comparing to nlohmann's jsonformoderncpp, we realized that `H_SURROGATE_START` and `H_SURROGATE_END` should be checked for as well, like so:

```suggestion
            if (decoded >= H_SURROGATE_START && decoded <= H_SURROGATE_END)
```